lightning/src/ln/reorg_tests.rs

```diff
@@ -180,3 +188,204 @@ fn test_onchain_htlc_claim_reorg_remote_commitment() {
+	// channel close on nodes[0] side. We also use this to provide very basic testing of logic
+	// around freeing background events which store monitor updates during block_[dis]connected.
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let persister: test_utils::TestPersister;
+	let new_chain_monitor: test_utils::TestChainMonitor;
+	let nodes_0_deserialized: ChannelManager<EnforcingSigner, &test_utils::TestChainMonitor, &test_utils::TestBroadcaster, &test_utils::TestKeysInterface, &test_utils::TestFeeEstimator, &test_utils::TestLogger>;
+	let mut nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+	let chan_id = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known()).2;
+
+	let channel_state = nodes[0].node.channel_state.lock().unwrap();
+	assert_eq!(channel_state.by_id.len(), 1);
+	assert_eq!(channel_state.short_to_id.len(), 1);
+	mem::drop(channel_state);
+
+	let mut headers = Vec::new();
+	let mut header = BlockHeader { version: 0x20000000, prev_blockhash: Default::default(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+	headers.push(header.clone());
+	for _i in 2..100 {
+		header = BlockHeader { version: 0x20000000, prev_blockhash: header.block_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		headers.push(header.clone());
+	}
+	if !reorg_after_reload {
+		while !headers.is_empty() {
+			nodes[0].node.block_disconnected(&headers.pop().unwrap());
+		}
+		check_closed_broadcast!(nodes[0], false);
+		{
+			let channel_state = nodes[0].node.channel_state.lock().unwrap();
+			assert_eq!(channel_state.by_id.len(), 0);
+			assert_eq!(channel_state.short_to_id.len(), 0);
+		}
+	}
+
+	if reload_node {
+		// Since we currently have a background event pending, it's good to test that we survive a
+		// serialization roundtrip. Further, this tests the somewhat awkward edge-case of dropping
+		// the Channel object from the ChannelManager, but still having a monitor event pending for
+		// it when we go to deserialize, and then use the ChannelManager.
+		let nodes_0_serialized = nodes[0].node.encode();
+		let mut chan_0_monitor_serialized = test_utils::TestVecWriter(Vec::new());
+		nodes[0].chain_monitor.chain_monitor.monitors.read().unwrap().iter().next().unwrap().1.write(&mut chan_0_monitor_serialized).unwrap();
+
+		persister = test_utils::TestPersister::new();
+		let keys_manager = &chanmon_cfgs[0].keys_manager;
+		new_chain_monitor = test_utils::TestChainMonitor::new(Some(nodes[0].chain_source), nodes[0].tx_broadcaster.clone(), nodes[0].logger, node_cfgs[0].fee_estimator, &persister, keys_manager);
+		nodes[0].chain_monitor = &new_chain_monitor;
+		let mut chan_0_monitor_read = &chan_0_monitor_serialized.0[..];
+		let (_, mut chan_0_monitor) = <(BlockHash, ChannelMonitor<EnforcingSigner>)>::read(
```

Comment:
Looks like you'll need to update this to `Option(BlockHash)` as a result of the rebase.